src/NuGet.Clients/NuGet.PackageManagement.UI/Converters/InstalledVulnerabilitiesCountConverter.cs

```diff
@@ -0,0 +1,37 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Diagnostics;
+using System.Globalization;
+using System.Windows.Data;
+
+namespace NuGet.PackageManagement.UI
+{
+    public class InstalledVulnerabilitiesCountConverter : IValueConverter
```

Comment:
We don't need to create a converter simply for formatting a string using a resource. We should be able to use a MultiBinding with a StringFormat and reference the resource string.

Reply:
The problem with multi binding is that we cannot use conditionals inside it, the only way to make it "dynamic" depending on the vulnerabilities count is to use a converter when binding.

```xaml
<TextBlock>
  <TextBlock.Text>
    <MultiBinding Converter="{StaticResource CustomConverter}">
      <Binding Source="{x:Static local:Resources.Label_InstalledVulnerabilitiesCount}" />
      <Binding Path="InstaledVulnerabilitiesCount" />
    </MultiBinding>
  </TextBlock.Text>
</TextBlock>
```

or something like this

```xaml
<MultiBinding Converter="{StaticResource StringFormatConverter}">
    <Binding Source="{x:Static local:Resources.Label_InstalledVulnerabilitiesCount}" />
    <Binding Path="InstaledVulnerabilitiesCount"
             Converter="{StaticResource CustomConverter}"/>
</MultiBinding>
```

Reply:
I prefer the Converter over any inline XAML syntax as this converter can be tested, and IMO, is far easier to read.

Reply:
> The problem with multi binding is that we cannot use conditionals inside it

A data triggers can handle visibility based on the value.

Reply:
MultiBinding doesn't have an inner Style property so if we wanted to use a DataTrigger it would look like this:

```xaml
<TextBlock>
  <TextBlock.Style>
    <Style TargetType="TextBlock">
      <Setter Property="Text">
        <Setter.Value>
          <MultiBinding Converter="{StaticResource StringFormatConverter}">
            <Binding Source="{x:Static local:Resources.Label_InstalledVulnerabilitiesCountSingle}" />
            <Binding Path="InstalledVulnerabilitiesCount" />
          </MultiBinding>
        </Setter.Value>
      </Setter>
      <Style.Triggers>
        <DataTrigger Binding="{Binding InstalledVulnerabilitiesIsMoreThanZero}" Value="True">
          <Setter Property="Text">
            <Setter.Value>
              <MultiBinding Converter="{StaticResource StringFormatConverter}">
                <Binding Source="{x:Static local:Resources.Label_InstalledVulnerabilitiesCount}" />
                <Binding Path="InstalledVulnerabilitiesCount" />
              </MultiBinding>
            </Setter.Value>
          </Setter>
        </DataTrigger>
      </Style.Triggers>
    </Style>
  </TextBlock.Style>
</TextBlock>
```

Reply:
Looks good! Instead of needing an additional property on the ViewModel for the boolean, you could do the data trigger such that it sets it to the singular form if the value is 1, otherwise it would be plural (or 0, but without double checking, I'm assuming we're handling that with visibility. If not, you can also handle the 0 case easily).

Reply:
You can also use a data template instead of specifying it in-line. With that, you can create a generalized Style that can be applied to any case where we need a singular / plural differentiator.